### ISSUE TITLE
Prevent infinite loop on method simulation & type resolution

### ISSUE THREAD

**Issue by jeff9finger** [sdaschner/jaxrs-analyzer#24]
When executing this plugin (version 0.6) on our jax-rs api, I get a java.lang.StackOverflowError.

Exception in thread "main" java.lang.StackOverflowError
	at java.util.Vector.<init>(Vector.java:148)
	at java.util.Vector.<init>(Vector.java:157)
	at java.util.Stack.<init>(Stack.java:53)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.MethodSimulator.<init>(MethodSimulator.java:45)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.InjectableArgumentMethodSimulator.<init>(InjectableArgumentMethodSimulator.java:33)
	at com.sebastian_daschner.jaxrs_analyzer.model.methods.ProjectMethod.invoke(ProjectMethod.java:47)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.MethodSimulator.simulateInvoke(MethodSimulator.java:177)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.MethodSimulator.simulate(MethodSimulator.java:94)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.MethodSimulator$$Lambda$199/975239896.accept(Unknown Source)
	at java.util.LinkedList$LLSpliterator.forEachRemaining(LinkedList.java:1235)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.MethodSimulator.simulateInternal(MethodSimulator.java:74)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.InjectableArgumentMethodSimulator.simulate(InjectableArgumentMethodSimulator.java:50)
...
	at com.sebastian_daschner.jaxrs_analyzer.model.methods.ProjectMethod.invoke(ProjectMethod.java:47)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.MethodSimulator.simulateInvoke(MethodSimulator.java:177)
	at com.sebastian_daschner.jaxrs_analyzer.analysis.bytecode.simulation.MethodSimulator.simulate(MethodSimulator.java:94)

**Comment by sdaschner** [comment]
Thanks!
Already working on a solution...

**Comment by jeff9finger** [comment]
Awesome! Thanks. If I can test something for you, please let me know.

**Comment by sdaschner** [comment]
Fixed in bcd0f0c and 9caf038. Feel free to test the latest snapshot version `0.7-SNAPSHOT`.

Therefore the Sonatype Snapshot repository needs to be included:

```
<pluginRepositories>
    <pluginRepository>
        <id>ossrh-snapshots</id>
        <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
        <releases>
            <enabled>false</enabled>
        </releases>
        <snapshots>
            <enabled>true</enabled>
        </snapshots>
    </pluginRepository>
</pluginRepositories>
```
